internal/webhook/template_webhook.go

```diff
@@ -119,7 +120,39 @@ func (*ServiceTemplateValidator) ValidateUpdate(_ context.Context, _ runtime.Obj
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
-func (*ServiceTemplateValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+func (in *ServiceTemplateValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	tmpl, ok := obj.(*v1alpha1.ServiceTemplate)
+	if !ok {
+		return admission.Warnings{"Wrong object"}, apierrors.NewBadRequest(fmt.Sprintf("expected ServiceTemplate but got a %T", obj))
+	}
+
+	var token string
+
+	for {
```

Comment:
the pattern looks odd to me, it definitely correct but I still have a question: why don't we just list everything as a single API call? the `delete` event isn't expected to happen frequently

Reply:
So I want to make sure that the `ServiceTemplate` object that the user calls delete on is not used by any `ManagedCluster` objects. I don't know how many `ManagedCluster` objects there could be in the management cluster, so therefore, I thought it is safer (w.r.t memory) to use pagination to list over all `ManagedCluster` objects.